### PR TITLE
Add an example of updating modal

### PR DESCRIPTION
**Pull request by KouWakai**
Hi!! I added updating modal example

since I wanted to know how could I handle updating modal in examples.

### CODE REVIEW

File: examples/modal/modal.go [hunk 1]
```diff
@@ -149,6 +173,22 @@ func handleModal(w http.ResponseWriter, r *http.Request) {
 		w.WriteHeader(http.StatusUnauthorized)
 		return
 	}
+
+	// update modal sample
+	switch i.Type {
+	//update when interaction type is view_submission
+	case slack.InteractionTypeViewSubmission:
+		api := slack.New("YOUR_TOKEN_HERE")
```
Comment (kanata2) [inline examples/modal/modal.go]: Already defined in L167.

File: examples/modal/modal.go [hunk 2]
```diff
@@ -149,6 +173,22 @@ func handleModal(w http.ResponseWriter, r *http.Request) {
...
+		//you can use any modal you want to show to users just like creating modal.
+		updateModal := updateModal()
+		//you can use hash and ViewID from payload and externalID if exists
```
Comment (kanata2) [inline examples/modal/modal.go]:
```suggestion
		// You must set one of external_id or view_id and you can use hash for avoiding race condition.
		// More details: https://api.slack.com/surfaces/modals/using#updating_apis
```

File: examples/modal/modal.go [hunk 3]
```diff
@@ -149,6 +173,22 @@ func handleModal(w http.ResponseWriter, r *http.Request) {
...
+		_, err := api.UpdateView(updateModal, "", i.View.Hash, i.View.ID)
+		if err != nil {
+			fmt.Printf("Error updating view: %s", err)
+			w.WriteHeader(http.StatusUnauthorized)
```
Comment (kanata2) [inline examples/modal/modal.go]:
```suggestion
			w.WriteHeader(http.StatusInternalServerError)
```

**Comment by KouWakai** [comment]
Thank you for feedback:pray:
I would modify codes to the suggestion

File: examples/modal/modal.go (hunk 1 above)
Comment (KouWakai) [reply examples/modal/modal.go]: Sure,I would delete L181.

File: examples/modal/modal.go (hunk 3 above)
Comment (KouWakai) [reply examples/modal/modal.go]: I see,this should be StatusInternalServerError.

File: examples/modal/modal.go [hunk 4]
```diff
@@ -149,6 +173,22 @@ func handleModal(w http.ResponseWriter, r *http.Request) {
 		w.WriteHeader(http.StatusUnauthorized)
 		return
 	}
+
+	// update modal sample
+	switch i.Type {
+	//update when interaction type is view_submission
+	case slack.InteractionTypeViewSubmission:
```
Comment (charlesoconor) [inline examples/modal/modal.go:172]: I couldn't get `api.UpdateView` to work under this event. I had to use `slack.InteractionTypeBlockActions` e.g `block_action`. Is this working for you?

Comment (KouWakai) [reply examples/modal/modal.go:172]: Thank you for the feedback,I would check that.

Comment (KouWakai) [reply examples/modal/modal.go:172]: It's working but modal closes right after update.

I think this is caused by slack server.
after update modal,app respond http status 200 to slack server and the server see it as over

I'v been running this code on local and connect via ngrok and this happen

Comment (KouWakai) [reply examples/modal/modal.go:172]: by adding `time.Sleep(time.Second * 2)` after api.UpdateView you can prevent sending http 200 and can see updated modal.

```go
	case slack.InteractionTypeViewSubmission:
		//you can use any modal you want to show to users just like creating modal.
		updateModal := updateModal()
		// You must set one of external_id or view_id and you can use hash for avoiding race condition.
		// More details: https://api.slack.com/surfaces/modals/using#updating_apis
		_, err := api.UpdateView(updateModal, "", i.View.Hash, i.View.ID)
		time.Sleep(time.Second * 2)
		if err != nil {
			fmt.Printf("Error updating view: %s", err)
			w.WriteHeader(http.StatusInternalServerError)
			return
		}
```

Comment (KouWakai) [reply examples/modal/modal.go:172]: @kanata2 @charlesoconor 
sorry for kept you waiting,could you review this code?

**Review by parsley42 — APPROVED** [review]
Even if this example isn't perfect, it's an expansion of what we had before.

**Pull request merged**